# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../mpfr/hotfix.patch.mips64
# Copyright (C) 2010 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -ur mpfr-2.4.2/mpfr-longlong.h mpfr-2.4.2.fix/mpfr-longlong.h
--- mpfr-2.4.2/mpfr-longlong.h	2009-11-29 21:43:08.000000000 -0500
+++ mpfr-2.4.2.fix/mpfr-longlong.h	2010-02-08 13:47:25.000000000 -0500
@@ -1023,6 +1023,11 @@
 #define UDIV_TIME 100
 #endif /* __mips */
 
+#if 0 /* JLR */
+Get rid of this to get rid of:
+mul.c:315: error: impossible constraint in 'asm'
+in T2 Mips64 build
+*/
 #if (defined (__mips) && __mips >= 3) && W_TYPE_SIZE == 64
 #if __GNUC__ > 2 || __GNUC_MINOR__ >= 7
 #define umul_ppmm(w1, w0, u, v) \
@@ -1035,6 +1040,7 @@
 #define UMUL_TIME 20
 #define UDIV_TIME 140
 #endif /* __mips */
+#endif /* JLR */
 
 #if defined (__ns32000__) && W_TYPE_SIZE == 32
 #define umul_ppmm(w1, w0, u, v) \
